src/apps/demo-app/src/app/github-issues/split/split.directive.ts

```diff
@@ -62,24 +60,17 @@ export class SplitDirective implements AfterContentInit, OnDestroy {
    * Use the pixel delta change to recalculate the area size (%)
    * Note: flex value may be '', %, px, or '<grow> <shrink> <basis>'
    */
-  calculateSize(value: string|number, delta: number) {
+  calculateSize(value: string, delta: number): number {
     const containerSizePx = this.elementRef.nativeElement.clientWidth;
-    const elementSizePx = Math.round(this.valueToPixel(value, containerSizePx));
+    const elementSizePx = Math.round(valueToPixel(value, containerSizePx));
 
     const elementSize = ((elementSizePx + delta) / containerSizePx) * 100;
     return Math.round(elementSize * 100) / 100;
   }
+}
 
-  /**
-   * Convert the pixel or percentage value to a raw
-   * pixel float value.
-   */
-  valueToPixel(value: string | number, parentWidth: number): number {
-    const isPercent = () => String(value).indexOf('px') < 0;
-    let size = parseFloat(String(value));
-    if (isPercent()) {
-      size = parentWidth * (size / 100);  // Convert percentage to actual pixel float value
-    }
-    return size;
-  }
+/** Convert the pixel or percentage value to a raw pixel float value */
+function valueToPixel(value: string, parentWidth: number): number {
```

Comment:
This does not work. Consider `valueToPixel('327')` which returns **NaN**.
You need
```ts
function valueToPixel(value: string, parentWidth = 100): number {
   ...
}

Reply:
`parentWidth` is not optional though?

Reply:
True that... I still contend my version is more clear. It is a number and defaults to `100`.